### PR TITLE
new: skip serialization if dns_domain is false

### PR DESCRIPTION
**Pull request by cloudguruab**
- Tested in unit and passes, ended up commenting out - not sure if this would even require its own testcase of any sorts?

### CODE REVIEW

**Comment by cloudguruab** [comment]
@jistr going to fail, but trying to implement a mock of serialized object params and sdk_params to tweak by emptying the dns_domain key and then checking its presence in both params, and sdk_params

File: os_migrate/tests/unit/test_network.py [hunk 1]
```diff
@@ -162,3 +163,24 @@ def test_network_sdk_params(self):
...
+    def test_network_serialization_when_dns_domain_is_false(self):
+        sdk_net = sdk_network()
+        net = Network.from_sdk(None, sdk_net)  # conn=None
+        params, info = net.params_and_info()
+        
+        with mock.patch.dict(params['dns_domain'], {'dns_domain': ''}):
+            self.assertEqual(params['dns_domain'], 'example.org')
+
+    def test_network_import_when_dns_domain_is_false(self):
+        net = Network.from_data(serialized_network())
```
Comment (jistr) [inline os_migrate/tests/unit/test_network.py:181]: I think we need to save the serialized_network() result to a variable, which will allow us to set the dns_domain to empty string, and only then we feed it to Network.from_data, and we can then validate that it is not present in sdk_params.

Comment (cloudguruab) [reply os_migrate/tests/unit/test_network.py:181]: Ahhhh looks like I was over complicating! - Thanks

**Comment by jistr** [comment]
Nice, let's just squash into a single `new` or `fix` commit (this is a user-facing change so it should appear in changelog) and we can merge. :+1:

**Comment by cloudguruab** [comment]
@jistr will merge documentation #546 first then rebase this branch ontop of that & merge

**Review by jistr — APPROVED** [review]

**Pull request merged**